gemd/util/impl.py

```diff
@@ -69,6 +69,18 @@ def _substitute(thing,
     # assert type(thing) == type(new), "{} is not {}".format(type(thing), type(new))
     return new
 
+def make_index(gems):
+    """
+    Given a list of GEMD objects, generates an index that can be used for substitute_links and substitute_objects methods.
+
+    :gems list: List of GEMD objects from which to create an index
+
+    """
+    gem_index = {}
+    for gem in gems:
+        for k, v in gem.uids.items():
+            gem_index[(k,v)] = gem
```

Comment:
Should we be invoking `set_uuids` here to guarantee that every object is mappable?

Reply:
I don't think we should. A person would never need an object within the index to be rehydrated if it never had a unique id; there would have been no scope/id pair to form the LinkByUID Object pair in the first place. Additionally, users use the "id" scoped id, traditionally, to keep track on registered versus non-registered objects.